webclient/app/src/app/features/apps/entities/entityDialog/EntityDialog.jsx

```diff
@@ -40,40 +42,36 @@ function EntityDialog(props) {
 
     useEffect(() => {
         if (entityId) {
-            setEntity({...entities.find(entity_ => entity_.id === entityId)});
+            const newEntity = {...entities.find(entity_ => entity_.id === entityId)};
+            newEntity.fields?.forEach(field => {
+                field.mandatory = field.fieldValidateRules?.includes("required")
+            })
+            setEntity(newEntity);
         } else {
             setEntity({...newEntity})
         }
     }, [entityId, entities])
 
     useEffect(() => {
-        if (!entity) {
-            return;
-        }
-        let hasError = false;
-
-        if (!RegexConfig.entityTitle.test(entity.name)) {
-            hasError = true;
-        }
+        setHasFormError(hasEntityFieldValidationError(entity));
+    }, [entity])
 
-        entity.fields?.forEach(field => {
-            if (!RegexConfig.fieldName.test(field.fieldName)) {
-                hasError = true;
-            }
+    function prepareClose() {
```

Comment:
warum wurde hier eine separate methode notwending? Wenn das refactoring ist - dafür gibt es ein ticket. Bitte nicht hier - dieser Teil ist ohne Änderung schon viel zu komplex. Das scheint mir auch nicht zu passen.

Reply:
Das ist das automatische Löschen, wenn das Feld leer ist. Das onClose reicht an sich nicht, weil dann auch immer die felder geprüft werden, ob sie leer sind oder nicht. onClose wird nur verwendet, um den Dialog an sich zu schließen. PrepareClose in dem Fall, wenn die Felder noch gesäubert werden sollen (Verhindert die leeren Felder auf der Ansicht).